corehq/apps/data_dictionary/views.py

```diff
@@ -92,10 +95,7 @@ def data_dictionary_json(request, domain, case_type_name=None):
                     ),
                     'name': prop.name,
                     'deprecated': prop.deprecated,
-                    'is_geo_case_property': (
-                        data_validation_enabled
-                        and prop.name == geo_case_property_name
-                    ),
+                    'is_geo_case_property': (prop.name == geo_case_property_name)
```

Comment:
(non blocking) how about passing the name of the geo case property separately and then just do this check in JS instead?
I'd prefer to keep the backend implementation straight forward and avoid modifying the collection to simply set a boolean.
I think it would be a bit simpler as well to do in JS instead.

Reply:
The validation has been moved to JS ([a91f8a5](https://github.com/dimagi/commcare-hq/pull/33426/commits/a91f8a58dd58a77cc4b33403af71db07bcf9f653)).